addons/markinglaser/XEH_postInit.sqf

```diff
@@ -1,45 +1,41 @@
 #include "script_component.hpp"
 #include "\a3\ui_f\hpp\defineDIKCodes.inc"
 
-// Events
-[QGVAR(laserOn),  LINKFUNC(onLaserOn)]  call CBA_fnc_addEventHandler;
-[QGVAR(laserOff), LINKFUNC(onLaserOff)] call CBA_fnc_addEventHandler;
+if (!hasInterface) exitWith {};
 
 // Keybinds
 ["ACE3 Vehicles", QGVAR(toggleLaser), LLSTRING(ToggleLaser), {
     // Ignore when in Zeus
-    if (!isNull curatorCamera) exitWith {};
+    if (!isNull curatorCamera) exitWith {false};
 
     private _vehicle = cameraOn;
     if !(_vehicle getVariable [QGVAR(enabled), false]) exitWith {false};
 
     private _controlledUnit = [ACE_player, ACE_controlledUAV # 1] select (unitIsUAV _vehicle);
 
-    private _canTurnOn = if (_vehicle getVariable [QGVAR(useTurret), false]) then {
-        private _turretInfo = _vehicle getVariable [QGVAR(turretInfo), []];
-        _controlledUnit == _vehicle turretUnit _turretInfo # 0
-    } else {
-        _controlledUnit == driver _vehicle
-    };
-
-    if (_canTurnOn) then {
-        if (_vehicle getVariable [QGVAR(laserOn), false]) then {
-            [QGVAR(laserOff), [_vehicle]] call CBA_fnc_globalEvent;
-        } else {
-            [QGVAR(laserOn), [_vehicle]] call CBA_fnc_globalEvent;
-        };
-
-        true
-    } else {
-        false
-    };
-}, "", [DIK_L, [false, false, true]]] call CBA_fnc_addKeybind;
-
-// JIP init
-if (didJIP) then {
-    {
-        if (_x getVariable [QGVAR(laserOn), false]) then {
-            [_x] call FUNC(onLaserOn);
-        };
-    } forEach (allMissionObjects "Air");
-};
+    private _turretInfo = _vehicle getVariable [QGVAR(turretInfo), []];
+    private _canTurnOn = _controlledUnit == _vehicle turretUnit _turretInfo;
+    if (!_canTurnOn) exitWith { false };
+
+    playSound "ACE_Sound_Click";
+    private _currentMode = _vehicle getVariable [QGVAR(laserMode), MODE_OFF];
+    private _newMode = (_currentMode + 1) % 3;
+    _vehicle setVariable [QGVAR(laserMode), _newMode, true];
+    true
+}, "", [DIK_L, [false, false, true]]] call CBA_fnc_addKeybind; // ALT-L
+
+["CBA_settingsInitialized", {
+    TRACE_1("settingsInitialized",1);
+
+    GVAR(pfEH) = -1;
+    ["visionMode", LINKFUNC(onLaserOn), true] call CBA_fnc_addPlayerEventHandler;
+    ["ace_laserOn", {
+        params ["", "_args"];
+        _args params ["_object"];
+        if !(_object getVariable [QGVAR(enabled), false]) exitWith {};
+        _object setVariable [QGVAR(smoothing), []];
+        _object setVariable [QGVAR(flashOffset), random 1]; // make flashes not syncronized
```

Comment:
```suggestion
        _object setVariable [QGVAR(flashOffset), random 1]; // make flashes not synchronized
```

Reply:
1) Don't we want to have the flashes be (more or less) globally synchronised? i.e. not having `random` run on each client
2) I assume you added this to avoid having 2 vehicles with synchronised flashing. What are the odds that, even without the `random`, it happens in practice?

Reply:
- it's blinking so fast that I don't think anyone will be able to tell the difference if it's actually global synced or not
- without the random every laser would be synced as they are all just using missionTime to do scripted blinking
```
        private _cycle = (CBA_missionTime + (_aircraft getVariable [QGVAR(flashOffset), 0])) % (1/3);
        _shouldBeOn = _cycle > 1/6